app/Http/Controllers/ProfileController.php

```diff
@@ -472,6 +472,15 @@ public function editCampSave(Request $request, $event_id)
         ]);
     }
 
+    // Setup payment for an existing registration
+    public function setupExistingPayment(Request $request, Event $event)
+    {
+        $participant = $request->user()->profile;
+        $payment = $participant->events()->findOrFail($event->id)->pivot->getPayment($existing = true);
```

Comment:
You cannot do `existing = true` this works because the side effect of an assignment is the value itself. 

I am also not a big fan of boolean arguments as it doesn't tell you much. Moreover you don't `get` a payment, you `make` a payment because the payment VO doesn't yet exist and you do not get is from the database or anything.

Why not so something like this (you can even go the extra mile and make the default value in the `existing` method true, because it should be.
```suggestion
$participant->events()->findOrFail($event->id)->pivot->makePayment()->existing(true);
``` 



With regards to the bool argument
```
        $payment = $participant->events()->findOrFail($event->id)->pivot->getPayment(existing: true);
```
or
```
        $payment = $participant->events()->findOrFail($event->id)->pivot->getPayment(true);
```